### PR TITLE
Header warning fixes

### PR DESCRIPTION
**Pull request by sgaist**
This pull request fixes the public headers include warnings generated when compiling the module as well as a private header missing "warning" warning.

### CODE REVIEW

**Comment by VSRonin** [comment]
I'm not completely sure about this. The whole thing is deployed to work as a Qt Module, make install does that.
`QT += xlsx`
Does the include and linking work for you with no include warnings.

I personally don't like to have external libraries work this way put the original author chose to do it this way

**Comment by sgaist** [comment]
If you mean, can I build an application without warning ? Yes. 

What I fixed with this PR is a series of warning generated when building a Qt module with "unclean" headers.

In the absolute, the standard Qt public headers are also written differently when using classes from other submodules (i.e. <QtModule/qclassname.h>) but I wanted to let that for another request.

This simplifies the life of people not using qmake to handle their project.

**Comment by topillar** [comment]
I made a shared lib fork--https://github.com/topillar/QtXlsxWriter, I don't like module way either, but it hard to follow other's commits.
